src/par/MESH_Partition.c

```diff
@@ -526,6 +612,231 @@ extern "C" {
 
 
 
+
+int MESH_Vol_Partition_Some_FN(Mesh_ptr mesh, int num_total, int *part, int begin, int end, Mesh_ptr *submeshes) {
+    int i, j, ic, idx, idx2, mpart_no;
+    int nv, nr, nrv, nre, nrf, nfe;
+    int grid=1, gfid=1, geid=1, gvid=1;
+    double xyz[3];
+    MVertex_ptr lmv, gmv;
+    MEdge_ptr lme, gme;
+    MFace_ptr lmf, gmf;
+    MRegion_ptr lmr, gmr;
+    List_ptr rverts, redges, rfaces, fedges, lmvlist, lmelist, lmflist, lmrlist;
+    MAttrib_ptr l2gatt, g2latt;
+    int part_index;
+
+    /* basic mesh information */
+    RepType rtype = MESH_RepType(mesh);
+    nv = MESH_Num_Vertices(mesh);
+    nr = MESH_Num_Regions(mesh);
+
+    int part_no;
+
+    g2latt = MESH_AttribByName(mesh,"Global2Local");
+
+    MFace_ptr *lrfaces = (MFace_ptr *) malloc(MAXPF3*sizeof(MFace_ptr));
+    int *lrfdirs = (int *) malloc(MAXPF3*sizeof(int));
+    MEdge_ptr *lfedges = (MEdge_ptr *) malloc(MAXPV2*sizeof(MEdge_ptr));
+    int *lfedirs = (int *) malloc(MAXPV2*sizeof(int));
+
+    /* for each partition */
+    for (part_no = begin; part_no < end; part_no++) {
+      part_index = part_no - begin;
+
+      l2gatt = MAttrib_New(submeshes[part_index],"Local2Global",POINTER,MALLTYPE);
+
+      idx = 0; ic = 0;
+      while ((gmr = MESH_Next_Region(mesh,&idx))) {
+        if (part[ic++] != part_no)
+          continue;
+	
+        rfaces = MR_Faces(gmr);
+        nrf = List_Num_Entries(rfaces);
+	
+        for (i = 0; i < nrf; i++) {
+          int found_lmf = 0;
+
+          gmf = List_Entry(rfaces,i);
+          MEnt_Get_AttVal(gmf,g2latt,0,0,&lmflist);
+          
+          if (!lmflist) {
+            lmflist = List_New(0);
+            MEnt_Set_AttVal(gmf,g2latt,0,0,lmflist);
+          }
+          else {
+            idx2 = 0;
+            while ((lmf = List_Next_Entry(lmflist,&idx2))) {
+              if (MF_Mesh(lmf) == submeshes[part_index]) {
+                found_lmf = 1;
+                break;
+              }
+            }
+          }
+
+          if (!found_lmf) {
+            lmf = MF_New(submeshes[part_index]);
+            MF_Set_GEntID(lmf,MF_GEntID(gmf));
+            MF_Set_GEntDim(lmf,MF_GEntDim(gmf));
+            mpart_no = MF_MasterParID(gmf);
+            if (mpart_no == 0) { /* not set */
+              MF_Set_MasterParID(lmf,part_no);
+              MF_Set_MasterParID(gmf,part_no+1);
```

Comment:
I think for what is being done in this PR, the master partition ID assignment will work as is. But as we are making changes to how we do partitioning and sending we may want to move the assignment of master partition ID on global mesh entities to a global step before we start making and sending submeshes off to different processors. Not sure why I did not do this before?